xscen/finalize.py

```diff
@@ -45,3 +50,166 @@ def change_units(ds: xr.Dataset, variables_and_units: dict):
+        Dictionary to pass to xscen.common.maybe_unstack fonction.
+        The format should be: {'coords': path_to_coord_file, 'rechunk': {'time': -1 }, 'stack_drop_nans': True}.
+    convert_calendar_kwargs: dict
+        Dictionary of argument to feed to xclim.core.calendar.convert_calendar. This will be the same for all variables.
+        If missing_by_vars is given, it will override the 'missing' argument given here.
+        Eg. {target': default, 'align_on': 'random'}
+    missing_by_var: list
+        Dictionary where the keys are the variables and the values are the argument to feed the `missing`
+        parameters of the xclim.core.calendar.convert_calendar for the given variable.
+        If missing_by_var == 'interpolate', the missing will be filled with NaNs, then linearly interpolated over time.
+    attrs_to_remove: dict
+        Dictionary where the keys are the variables and the values are a list of the attrs that should be removed.
+        For global attrs, use the key 'global'.
+        The element of the list can be exact matches for the attributes name
+        or use the same substring matching rules as intake_esm:
+        - ending with a '*' means checks if the substring is contained in the string
+        - starting with a '^' means check if the string starts with the substring.
+        eg. {'global': ['unnecessary note', 'cell*'], 'tasmax': 'old_name'}
+    remove_all_attrs_except: dict
+        Dictionary where the keys are the variables and the values are a list of the attrs that should NOT be removed,
+        all other attributes will be deleted. If None (default), nothing will be deleted.
+        For global attrs, use the key 'global'.
+        The element of the list can be exact matches for the attributes name
+        or use the same substring matching rules as intake_esm:
+        - ending with a '*' means checks if the substring is contained in the string
+        - starting with a '^' means check if the string starts with the substring.
+        eg. {'global': ['necessary note', '^cat/'], 'tasmax': 'new_name'}
+    add_attrs: dict
+        Dictionary where the keys are the variables and the values are a another dictionary of attributes.
+        For global attrs, use the key 'global'.
+        eg. {'global': {'title': 'amazing new dataset'}, 'tasmax': {'note': 'important info about tasmax'}}
+    change_attr_prefix: str
+        Replace "cat/" in the catalogue global attrs by this new string
+    to_level: str
+        The processing level to assign to the output.
+
+    Returns
+    -------
+    ds: xr.Dataset
+        Cleaned up dataset
+    """
+
+    if variables_and_units:
+        logger.info(f"Converting units: {variables_and_units}")
+        ds = change_units(ds=ds, variables_and_units=variables_and_units)
+
+    # unstack nans
+    if maybe_unstack_dict:
+        ds = maybe_unstack(ds, **maybe_unstack_dict)
+
+    # convert calendar
+    if convert_calendar_kwargs:
+        ds_copy = ds.copy()
```

Comment:
Is it safe to do, memory-wise?

Reply:
I think so. We don't load it in memory from what I understand of lazy computing, it will only compute it when we need it.